### PR TITLE
Implement sub functions

### PR DESCRIPTION
**Pull request by quisar**

### CODE REVIEW

File: compile.ts [hunk 1]
```diff
@@ -575,7 +630,24 @@ class Compiler {
   ): Variable {
     let dest = outs[0] || this.#temp();
     const args: (string | Variable)[] = [];
-    const info = methods[name];
+    let info = methods[name];
+    if (!info && this.subs.has(name)) {
```
Comment (ribrdb) [inline compile.ts:634]: You're also supposed to be able to recursively call the "main" function of the behavior. Maybe we should add the exported function to subs also, and keep track of which one is the main?

File: methods.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface MethodInfo {
   loop?: boolean;
   special?: 'txt'|'bp';
   c?: number;
+  sub?: string;
```
Comment (ribrdb) [inline methods.ts:11]: Can you also add this to `scripts/geninstr.ts`, line 383.

**Review by ribrdb — CHANGES_REQUESTED** [review]

File: methods.ts (hunk 2 above)
Comment (quisar) [reply methods.ts:11]: Done. Can you document how to use this script?

Comment (ribrdb) [reply methods.ts:11]: It's kind of complicated. You need to extract `instructions.lua` from the game's main mod, then run it through extractor.cjs to convert it to json. However I think I also had to hand patch luaparse to handle lua 5.4 syntax

**Pull request merged**